### DISCUSSION TITLE
Enforcing Rotational Symmetry Constraint

### DISCUSSION THREAD

**Discussion by lbullard-c2** [mdolab/dafoam#473] [General]
Due to the lack of cyclicAMI support, I am attempting to optimize the performance of a propeller by deforming the entire geometry. While the optimization cycle is overall effective, it tends to deform each blade slightly differently, or asymmetrically deform the hub.

Is it possible to either enforce rotational symmetry in the optimization?

Another approach I have considered is giving each blade a FFD bounding box - is it then possible to deform each one the same amount?

**Comment by raisePiginYourhome** [comment]
can you share the propeller shape picture, i want to learn if we have the same problem.

**Comment by friedenhe** [comment]
Check the NREL6 tutorial where we use two blocks of FFD for both blades. Refer to: https://github.com/DAFoam/tutorials/blob/main/NREL6_Wind_Turbine/runScript_v2.py You can set up linear constraints to link the displacement of FFDs.
